PWGUD/Tasks/upcJpsiCentralBarrelCorr.cxx

```diff
@@ -0,0 +1,870 @@
+    Correlation.add("Correlation/Electron/Coherent/Phi1", "Phi1", {HistType::kTH1F, {axisPhi}});
+    Correlation.add("Correlation/Electron/Coherent/Phi2", "Phi2", {HistType::kTH1F, {axisPhi}});
+    Correlation.add("Correlation/Electron/Coherent/CosThetaPhi", "CosThetaPhi", {HistType::kTH2F, {{axisAngTheta}, {axisPhi}}});
+
+    Correlation.add("Correlation/Electron/Incoherent/AccoplAngle", "AccoplAngle", {HistType::kTH1F, {axisAccAngle}});
+    Correlation.add("Correlation/Electron/Incoherent/CosTheta", "CosTheta", {HistType::kTH1F, {axisAngTheta}});
+    Correlation.add("Correlation/Electron/Incoherent/Phi", "Phi", {HistType::kTH1F, {axisPhi}});
+    Correlation.add("Correlation/Electron/Incoherent/Phi1", "Phi1", {HistType::kTH1F, {axisPhi}});
+    Correlation.add("Correlation/Electron/Incoherent/Phi2", "Phi2", {HistType::kTH1F, {axisPhi}});
+    Correlation.add("Correlation/Electron/Incoherent/CosThetaPhi", "CosThetaPhi", {HistType::kTH2F, {{axisAngTheta}, {axisPhi}}});
+
+    // Asymmetry histograms
+    Asymmetry.add("Asymmetry/Muon/Coherent/DeltaPhi", "DeltaPhi", {HistType::kTH1F, {{180, -PI, PI}}});
+    Asymmetry.add("Asymmetry/Muon/Incoherent/DeltaPhi", "DeltaPhi", {HistType::kTH1F, {{180, -PI, PI}}});
+    Asymmetry.add("Asymmetry/Electron/Coherent/DeltaPhi", "DeltaPhi", {HistType::kTH1F, {{180, -PI, PI}}});
+    Asymmetry.add("Asymmetry/Electron/Incoherent/DeltaPhi", "DeltaPhi", {HistType::kTH1F, {{180, -PI, PI}}});
+  }
+
+  float particleMass(TDatabasePDG* pdg, int pid)
+  {
+    auto mass = 0.;
+    TParticlePDG* pdgparticle = pdg->GetParticle(pid);
+    if (pdgparticle != nullptr) {
+      mass = pdgparticle->Mass();
+    }
+    return mass;
+  }
+
+  template <typename T>
+  bool GoodTrackCuts(T const& track)
+  {
+    // kinematics
+    if (RecoDecay::eta(std::array{track.px(), track.py(), track.pz()}) < -EtaCut || RecoDecay::eta(std::array{track.px(), track.py(), track.pz()}) > EtaCut) {
+      return false;
+    }
+    // DCA
+    if (track.dcaZ() > dcaZCut) {
+      return false;
+    }
+    if (track.dcaXY() > dcaXYCut) {
+      return false;
+    }
+    // ITS
+    if (!track.hasITS()) {
+      return false;
+    }
+    if (track.itsNCls() < ITSNClsCut) {
+      return false;
+    }
+    if (track.itsChi2NCl() > ITSChi2NClsCut) {
+      return false;
+    }
+    //  TPC
+    if (!track.hasTPC()) {
+      return false;
+    }
+    if (track.tpcNClsCrossedRows() < TPCNClsCrossedRowsCut) {
+      return false;
+    }
+    if (track.tpcChi2NCl() > TPCChi2NCls) {
+      return false; // TPC chi2
+    }
+
+    return true;
+  }
+
+  template <typename C>
+  bool CandidateCuts(C const& mother)
+  {
+    if (abs(RecoDecay::y(std::array{mother[0], mother[1], mother[2]}, mother[3])) > 0.8f) {
+      return false;
+    }
+
+    double energy = RecoDecay::e(mother[0], mother[1], mother[2], mother[3]);
+    if (RecoDecay::m(std::array{mother[0], mother[1], mother[2]}, energy) < 2.5f) {
```

Comment:
What are you trying to calculate here?

Reply:
I am trying to calculate the mass to select only mother particles over 2,5 GeV.

Reply:
It seems to me that you are just unnecessarily calculating what you already have in `mother[3]`.